src/services/dealer-price/index.ts

```diff
@@ -0,0 +1,122 @@
+import util from "util"
+
+import { credentials } from "@grpc/grpc-js"
+
+import { UnknownDealerPriceServiceError } from "@domain/dealer-price"
+
+import { baseLogger } from "../logger"
+
+import { PriceServiceClient } from "./proto/services/price/v1/price_service_grpc_pb"
+import {
+  GetExchangeRateForImmediateUsdBuyRequest,
+  GetExchangeRateForImmediateUsdBuyResponse,
+  GetExchangeRateForImmediateUsdSellRequest,
+  GetExchangeRateForImmediateUsdSellResponse,
+  GetExchangeRateForFutureUsdBuyRequest,
+  GetExchangeRateForFutureUsdBuyResponse,
+  GetExchangeRateForFutureUsdSellRequest,
+  GetExchangeRateForFutureUsdSellResponse,
+} from "./proto/services/price/v1/price_service_pb"
+
+const serverPort = process.env.PRICE_SERVER_PORT ?? "50055"
+const client = new PriceServiceClient(
+  `localhost:${serverPort}`,
+  credentials.createInsecure(),
+)
+
+const clientGetExchangeRateForImmediateUsdBuy = util.promisify(
+  client.getExchangeRateForImmediateUsdBuy,
+)
+const clientGetExchangeRateForImmediateUsdSell = util.promisify(
+  client.getExchangeRateForImmediateUsdSell,
+)
+const clientGetExchangeRateForFutureUsdBuy = util.promisify(
+  client.getExchangeRateForFutureUsdBuy,
+)
+const clientGetExchangeRateForFutureUsdSell = util.promisify(
+  client.getExchangeRateForFutureUsdSell,
+)
+
+export const DealerPriceService = (): IDealerPriceService => {
+  const getExchangeRateForImmediateUsdBuy = async function (
+    amountInSatoshis: Satoshis,
+  ): Promise<UsdCents | DealerPriceServiceError> {
+    try {
+      const response = (await clientGetExchangeRateForImmediateUsdBuy(
+        new GetExchangeRateForImmediateUsdBuyRequest().setAmountInSatoshis(
+          amountInSatoshis,
+        ),
+      )) as GetExchangeRateForImmediateUsdBuyResponse
+      return response.getPriceInUsd() as UsdCents
+    } catch (error) {
+      baseLogger.error(
+        { error },
+        "GetExchangeRateForImmediateUsdBuy unable to fetch price",
+      )
+      return new UnknownDealerPriceServiceError()
+    }
+  }
+
+  const getExchangeRateForImmediateUsdSell = async function (
+    amountInUsd: UsdCents,
+  ): Promise<Satoshis | DealerPriceServiceError> {
+    try {
+      const response = (await clientGetExchangeRateForImmediateUsdSell(
+        new GetExchangeRateForImmediateUsdSellRequest().setAmountInUsd(amountInUsd),
+      )) as GetExchangeRateForImmediateUsdSellResponse
+      return response.getPriceInSatoshis() as Satoshis
+    } catch (error) {
+      baseLogger.error(
+        { error },
+        "GetExchangeRateForImmediateUsdBuy unable to fetch price",
+      )
+      return new UnknownDealerPriceServiceError()
+    }
+  }
+
+  const getExchangeRateForFutureUsdBuy = async function (
```

Comment:
should we use the wording Option instead of Future? Option hint at the optionality of the transaction. it's not like a `forward` contract where the transaction will happen in the future no matter what.

Reply:
I'm not using the instrument's name "Futures" here, just saying it is for *later* settlement. Option/Forward/Futures are too specific and the action is really about getting the result of conversion according to a rate for now or later...
If you have a better wording, I'm ready to change them all.

Reply:
my point is that: `GetExchangeRateForImmediateUsdBuyRequest` is like a market order on how it will be handle in the backend. 

the dealer can assume that every time this function is been called, there will be an associated transaction

for the `getExchangeRateForFutureUsdBuy`, it's really a quote. there won't be necessarily a transaction associated to it. 

whether it's `quote` or `option` or something else, I'm not sure, but here the wording doens't really show the different between the 2 types of request (in my opinion at least) which is why I like the `option`. maybe you prefer `quote`?

Reply:
Yeah not sure, I just wanted to convey that one is a request for the immediate usd buy/sell "exchange rate" or quote while the other is for a future/later/delayed buy/sell "exchange rate" or quote and saying Option like the instrument is kind of leaking the implementation.
The original naming was confusing me so I changed them but really don't mind swapping for something else.
Just leave the 4 names in the comment and I will replace them.